controller/trackerquery.go

```diff
@@ -93,12 +93,15 @@ func (c *TrackerqueryController) Create(ctx *app.CreateTrackerqueryContext) erro
 	}
 	err = application.Transactional(c.db, func(appl application.Application) error {
 		trackerQuery := remoteworkitem.TrackerQuery{
-			Query:     ctx.Payload.Data.Attributes.Query,
-			Schedule:  ctx.Payload.Data.Attributes.Schedule,
-			TrackerID: ctx.Payload.Data.Relationships.Tracker.Data.ID,
-			SpaceID:   *ctx.Payload.Data.Relationships.Space.Data.ID,
+			Query:          ctx.Payload.Data.Attributes.Query,
+			Schedule:       ctx.Payload.Data.Attributes.Schedule,
+			TrackerID:      ctx.Payload.Data.Relationships.Tracker.Data.ID,
+			SpaceID:        *ctx.Payload.Data.Relationships.Space.Data.ID,
+			WorkItemTypeID: ctx.Payload.Data.Relationships.BaseType.Data.ID,
+		}
+		if ctx.Payload.Data.ID != nil {
+			trackerQuery.ID = *ctx.Payload.Data.ID
```

Comment:
I wonder whether the UUID should be validated.

Reply:
@baijum are you talking about a check if the `id` contains string or uuid?

Reply:
Check whether the value of `*ctx.Payload.Data.ID` is real UUID (It's not a string type)

Reply:
I mean what if the value is some random string. It will reach the database layer and fail. How about failing early?

Reply:
@baijum At design level itself its specified that the value should be a UUID (https://github.com/fabric8-services/fabric8-wit/blob/master/design/tracker_queries.go#L13)